### PR TITLE
correctly define $LIBFFT_MT for Intel MKL

### PR DESCRIPTION
**Pull request by boegel**
fix for #2672

@akesandgren I'm not 100% sure this is correct, so please check.
I assume you have a particular use case in mind for this?

### CODE REVIEW

**Comment by boegel** [comment]
@akesandgren Tested with `QuantumESPRESSO-6.2-intel-2017b.eb` and `QuantumESPRESSO-6.3-intel-2018b.eb` where `$LIBFFT_MT` is picked up, seems to work (it builds without issue, don't know how to test QE).

File: test/framework/toolchain.py [hunk 1]
```diff
@@ -870,9 +870,16 @@ def test_old_new_iccifort(self):
...
         # no -lguide
+        libblas_intel4 = "-Wl,-Bstatic -Wl,--start-group -lmkl_intel_lp64 -lmkl_sequential -lmkl_core"
+        libblas_intel4 += " -Wl,--end-group -Wl,-Bdynamic"
         libblas_mt_intel4 = "-Wl,-Bstatic -Wl,--start-group -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core"
         libblas_mt_intel4 += " -Wl,--end-group -Wl,-Bdynamic -liomp5 -lpthread"
 
+        libfft_intel4 = libblas_intel4.replace('-lmkl_intel_lp64', '-lfftw3xc_intel -lmkl_intel_lp64')
+        libfft_mt_intel4 = libblas_mt_intel4.replace('-lmkl_intel_lp64', '-lfftw3xc_intel -lmkl_intel_lp64')
+        libfft_mt_intel4 = "-Wl,-Bstatic -Wl,--start-group -lfftw3xc_intel -lmkl_intel_lp64 -lmkl_intel_thread"
+        libfft_mt_intel4 += " -lmkl_core -liomp5 -lpthread -Wl,--end-group -Wl,-Bdynamic"
```
Comment (akesandgren) [inline test/framework/toolchain.py]: These two lines doesn't make sense, you're overwriting the value you just set libfft_mt_intel4 to.

Comment (akesandgren) [reply test/framework/toolchain.py]: Hmmm, I think the replace would be more correct? I.e. just as libfft_intel4 uses replace on libblas, libfft_mt_intel4 should use replace on libblas_mt ?? Shorter and the -liomp5 -lpthread should be outside the grouping as far as I can see from the code in toolchain/fft.py above

Comment (boegel) [reply test/framework/toolchain.py]: Agreed, and fixed in 269d6ef

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**